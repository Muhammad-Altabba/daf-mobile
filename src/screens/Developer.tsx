/**
 * Serto Mobile App
 *
 */
import * as React from 'react'
import { NavigationScreenProps } from 'react-navigation'
import { useTranslation } from 'react-i18next'
import { Switch } from 'react-native'
import {
  Container,
  Text,
  Button,
  Constants,
  Screen,
  ListItem,
  Section,
} from '@kancha/kancha-ui'
import { Colors } from '../theme'
import { Screens } from '../navigators'
import Config from 'react-native-config'

export default ({ navigation }: NavigationScreenProps) => {
  const { t, i18n } = useTranslation()
  return (
    <Screen scrollEnabled={true} safeArea={true}>
      <Container backgroundColor={Colors.BRAND} padding alignItems={'center'}>
        <Text bold textColor={Colors.WHITE}>
          {t('Environment')}: {Config.ENV}
        </Text>
      </Container>

      <Container>
        <Section title={'Developer tooling'}>
          <ListItem onPress={() => navigation.navigate(Screens.Logs.screen)}>
            {t('Logs')}
          </ListItem>
          <ListItem onPress={() => navigation.navigate(Screens.Config.screen)}>
            {t('Configuration')}
          </ListItem>
          <ListItem onPress={() => navigation.navigate(Screens.Signer.screen)}>
            {t('Signer')}
          </ListItem>
          <ListItem
            onPress={() => navigation.navigate(Screens.Codepush.screen)}
          >
            {t('CodePush')}
          </ListItem>
          <ListItem
            last
            onPress={() => navigation.navigate(Screens.Crash.screen)}
          >
            {t('CrashReporting')}
          </ListItem>
          {/* <ListItem onPress={() => navigation.navigate(Screens.Locale.screen)}>{t('Localisation')}</ListItem> */}
        </Section>

        <Section title={'Language'}>
          <ListItem
            last
            iconLeft={
              <Switch
                value={i18n.language === 'es'}
                onValueChange={() =>
                  i18n.changeLanguage(i18n.language === 'es' ? 'en' : 'es')
                }
              />
            }
          >
            {i18n.language === 'en' ? t('English') : t('Spanish')}
          </ListItem>
        </Section>
      </Container>

      <Container flex={1} padding>
        {/* <Container marginBottom>
          <Button
            fullWidth
            type={Constants.BrandOptions.Primary}
            block={Constants.ButtonBlocks.Filled}
            buttonText={t('Logs')}
            onPress={() => navigation.navigate('Logs')}
          />
        </Container>
        <Container marginBottom>
          <Button
            fullWidth
            type={Constants.BrandOptions.Primary}
            block={Constants.ButtonBlocks.Filled}
            buttonText={t('Configuration')}
            onPress={() => navigation.navigate('Config')}
          />
        </Container>
        <Container marginBottom>
          <Button
            fullWidth
            type={Constants.BrandOptions.Primary}
            block={Constants.ButtonBlocks.Filled}
            buttonText={t('Signer')}
            onPress={() => navigation.navigate('Signer')}
          />
        </Container>
        <Container marginBottom>
          <Button
            fullWidth
            type={Constants.BrandOptions.Primary}
            block={Constants.ButtonBlocks.Filled}
            buttonText={t('CodePush')}
            onPress={() => navigation.navigate('Codepush')}
          />
        </Container>
        <Container marginBottom>
          <Button
            fullWidth
            type={Constants.BrandOptions.Primary}
            block={Constants.ButtonBlocks.Filled}
            buttonText={t('change')}
            onPress={() =>
              i18n.changeLanguage(i18n.language === 'es' ? 'en' : 'es')
            }
          />
        </Container>
        <Container marginBottom>
          <Button
            fullWidth
            type={Constants.BrandOptions.Warning}
            block={Constants.ButtonBlocks.Outlined}
            buttonText={'Crash App 6'}
            onPress={() => {
              throw new Error('Sample error from JS 6')
            }}
          />
        </Container> */}
      </Container>
    </Screen>
  )
}
