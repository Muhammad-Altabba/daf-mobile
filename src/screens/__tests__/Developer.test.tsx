import 'react-native'
import React from 'react'
import Developer from '../Developer'
import { render } from 'react-native-testing-library'

const navigation = {
  navigate: jest.fn(),
}

it('renders correctly', () => {
  // @ts-ignore
  const tree = render(<Developer navigation={navigation} />).toJSON()
  expect(tree).toMatchSnapshot()
})
