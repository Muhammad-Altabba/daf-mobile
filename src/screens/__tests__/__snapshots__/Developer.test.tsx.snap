// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />,
  <RCTScrollView
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    contentInsetAdjustmentBehavior="never"
    getTextInputRefs={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onScroll={[Function]}
    scrollEventThrottle={200}
    scrollToBottomOnKBShow={false}
    scrollToInputAdditionalOffset={75}
    startScrolledToBottom={false}
    style={
      Object {
        "backgroundColor": "#EFEEF3",
      }
    }
  >
    <View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": 1,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": 1,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0078ff",
                "borderRadius": 8,
                "maxWidth": 300,
                "opacity": 1,
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 20,
                }
              }
            >
              Logs
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0078ff",
                "borderRadius": 8,
                "maxWidth": 300,
                "opacity": 1,
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 20,
                }
              }
            >
              Signer
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0078ff",
                "borderRadius": 8,
                "maxWidth": 300,
                "opacity": 1,
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 20,
                }
              }
            >
              CodePush
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0078ff",
                "borderRadius": 8,
                "maxWidth": 300,
                "padding": 15,
              }
            }
          >
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": undefined,
                  "borderRadius": undefined,
                  "flex": undefined,
                  "flexDirection": "row",
                  "height": undefined,
                  "justifyContent": "center",
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": undefined,
                    "borderRadius": undefined,
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": undefined,
                    "justifyContent": "center",
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "width": undefined,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                    }
                  }
                >
                  change
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#333333",
                "fontSize": 24,
              }
            }
          >
            Configuration
          </Text>
          <Text
            style={
              Object {
                "color": "#333333",
                "fontSize": 16,
                "lineHeight": 22,
              }
            }
          >
            {"ENV":"dev","BRAND_COLOR":"#FF0000"}
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />,
]
`;
